larq_compute_engine/mlir/transforms/prepare_tf.cc

```diff
@@ -103,27 +103,17 @@ bool IsSamePadding(Attribute paddings_attr, Value input, Value output,
   auto output_shape = output.getType().cast<RankedTensorType>().getShape();
   auto strides = strides_attr.getValue();
 
-  int pad_height_left = paddings.getValue<int>({1, 0});
-  int pad_height_right = paddings.getValue<int>({1, 1});
-  int pad_width_left = paddings.getValue<int>({2, 0});
-  int pad_width_right = paddings.getValue<int>({2, 1});
-
-  int pad_height = pad_height_left + pad_height_right;
-  int pad_width = pad_width_left + pad_width_right;
-
   int stride_height = strides[1].cast<IntegerAttr>().getInt();
   int stride_width = strides[2].cast<IntegerAttr>().getInt();
 
-  return paddings.getValue<int>({0, 0}) == 0 &&
-         paddings.getValue<int>({0, 1}) == 0 &&
-         output_shape[1] == CeilDiv(input_shape[1], stride_height) &&
-         output_shape[2] == CeilDiv(input_shape[2], stride_width) &&
-         pad_height_left == pad_height / 2 &&
-         pad_height_right == (pad_height + 1) / 2 &&
-         pad_width_left == pad_width / 2 &&
-         pad_width_right == (pad_width + 1) / 2 &&
-         paddings.getValue<int>({3, 0}) == 0 &&
-         paddings.getValue<int>({3, 1}) == 0;
+  // Check that there is no padding in the batch and channel dimensions
+  // Functions defined in `padding.h`
+  return IsNoPadding(paddings, 0) &&
+         IsSamePadding1D(paddings, 1, input_shape[1], output_shape[1],
+                         stride_height) &&
+         IsSamePadding1D(paddings, 2, input_shape[2], output_shape[2],
+                         stride_width) &&
+         IsNoPadding(paddings, 3);
```

Comment:
Maybe this was already like this, but does this work with non 2D data? Here you assume the channel dimension is dimension 3, but what if there is only one spatial dimension? I guess then there is no Conv2D and this function doesn't trigger, is that it?

Reply:
This was indeed already like this, and it should indeed work because the function is only ever called if its followed by a (Depthwise)Conv2D. However I think it's a good idea to add some extra checks for this, I will do that.

Reply:
I've added some extra checks in [f28fe61](https://github.com/larq/compute-engine/pull/712/commits/f28fe619410e4b8efdc8045e58b059bf87d0028f)